### ISSUE TITLE
Grammar and wording checks installer.nsi

### ISSUE THREAD

**Issue by ann0see** [corrados/jamulus#861]
@gilgongo I think this is something for you. Could you please check the new [installer.nsi](https://github.com/corrados/jamulus/blob/master/windows/installer.nsi) file for language mistakes?

Furthermore, we should link an ASIO section on jamulus.io in the new installer.

**Comment by corrados** [comment]
This: https://github.com/corrados/jamulus/blob/1ce54ba0566c9064939f88ea13f35999cb97ae61/windows/installer.nsi#L31 is from the "old days of Jamulus". Should we keep it or use an updated text instead?

**Comment by ann0see** [comment]
I think this depends on what you think. The website uses "Make music online. With friends. For free." but I think it could be changed to "Jam online. With friends. For free" or maybe we could open a community voting?

**Comment by ann0see** [comment]
Also the ASIO detection warning must explain a bit more.

**Comment by dcorson-ticino-com** [comment]
I would go with "Make music".  I have noticed that may musicians around me (French part of Switzerland) don't know the term "to jam".  Thus they find the name Jamulus strange.
I thought "to jam" was well known and international, but apparently it isn't.

**Comment by corrados** [comment]
> The website uses "Make music online. With friends. For free." but I think it could be changed to "Jam online. With friends. For free"

I would use the same as on the homepage. If you change the text on the homepage, you should then change it in the installer as well.

> https://github.com/corrados/jamulus/blob/a3238e960741e08526d1482c2a5efc6e56bdd897/windows/installer.nsi#L92

I would not use "bug". I would remove the sentence "Most likely, this is due to a bug in the old installer of Jamulus.". If the users read about a bug right when they want to install Jamulus, that might confuse them. The sentence does not give a Jamulus user any useful information for him.

**Comment by ann0see** [comment]
Ok. The user should nevertheless know that it's 99% not a problem he introduced.

**Comment by corrados** [comment]
Well, he actually did introduce it by installing the old Jamulus version. Anyway, we should keep the text as short and informative as possible. The info we want to ship: For proper runtime behavior of the Jamulus software, it is recommended to uninstall the old Jamulus version.

**Comment by ann0see** [comment]
Yes. But the problem of this is that it is not only an old version of Jamulus but it is a wrongly installed version of Jamulus.

If one time somebody would introduce a 32Bit installer for 64Bit windows (I wouldn't really see why this would be needed but there was a feature request somewhere) this message should still work and explain the "issue".

**Comment by ann0see** [comment]
@corrados after thinking a little bit longer about this, we could remove the sentence with the bug since it already clearly tells the inexperienced user what to do.

**Comment by gilgongo** [comment]
Just having a look at this and it seems generally OK but it's hard to say without seeing the messages in actual context.

A couple of questions:

1. For OLD_VER_REMOVE_FAILED, I guess this would hardly ever happen, but if it did, what's the worst that would happen if we just silently install the new version and leave the old one in place?

2.  Referring to UI in messages is usually a bit unsafe (and in fact may not be accessible). So for example, in ASIO_EXIT_NO_DRIVER  instead of saying _"To return to this page, click 'No'. If you click 'Yes', the installation will continue."_ it's  recommended just to put the action text on each button. So instead of that message you would have two buttons, one saying  "Go back" and the other "Continue installation" perhaps.

**Comment by ann0see** [comment]
>  what's the worst that would happen if we just silently install the new version and leave the old one in place?

Not sure. I think it could confuse the user if he had two versions of Jamulus on his PC. I don't think Jamulus would break.

> So instead of that message you would have two buttons, one saying "Go back" and the other "Continue installation" perhaps.

I hope this is possible.

**Comment by gilgongo** [comment]
I guess if they could _launch_ the older version (so perhaps seeing two Jamulus icons in the start menu?), then that would be a problem. If manually uninstalling is simply a matter of deleting the .exe file from Programs though, then perhaps we should say that in the message?

**Comment by ann0see** [comment]
Concerning 2:

Just did a quick google search and I'm afraid that it's only possible by using a hack: https://stackoverflow.com/questions/46741179/nsis-message-box-custom-buttons

I would prefer to rephrase it to a yes/no question which makes clear that "No" refers to something like "Go back" ("Do you want to continue?")


>  If manually uninstalling is simply a matter of deleting the .exe file from Programs though, then perhaps we should say that in the message?

I think if this error comes up, we don't really know what went wrong and the user would need to do a lot more troubleshooting (= ask for help on the forum?)

**Comment by ann0see** [comment]
What about this: https://github.com/ann0see/jamulus/blob/344838838f035d623b74bdad65bc5b45c0feb109/windows/translate/en.nsi#L37

**Comment by gilgongo** [comment]
Not sure mentioning a bit of a technical term like "latency" is a best here, so perhaps:

"${APP_NAME} needs an ASIO audio driver to work, but we couldn't find one on your PC. You should install one like ASIO4ALL (More information on jamulus.io under Installation for Windows). Do you still want to continue with the installation of ${APP_NAME} first?"

Again, what's the reason for letting them continue?

**Comment by ann0see** [comment]
> Again, what's the reason for letting them continue?

Somebody might like to install an ASIO driver after the installation of Jamulus since he only wants to run a server.

I personally dislike forcing somebody into doing something which is not 100% needed.

**Comment by gilgongo** [comment]
OK sure. I would expect most people (often more naive users) simply don't read messages and want to progress if they can, so they probably will in this case. Whether the number of those users is more than the number of those who read the message and decide to progress (despite us not explaining why they would want to) is hard to say, 

But given the fact that the consequences of progressing in ignorance are not large (we simply repeat the messaging about needing to have an ASIO driver if they try to run Jamulus after ignoring the warning, I assume) then we can keep it as it is without much consequence.

**Comment by ann0see** [comment]
> we simply repeat the messaging about needing to have an ASIO driver if they try to run Jamulus after ignoring the warning

Exactly this message is enough in my opinion. 

Further argument:
If somebody only wants to run the server, there's no need for ASIO.

**Comment by ann0see** [comment]
@gilgongo can you edit the file with your proposal? If everything else is ok, we can close this issue.

**Comment by gilgongo** [comment]
@ann0see On which branch?

**Comment by ann0see** [comment]
It should be on master.

**Comment by gilgongo** [comment]
When I try switching to a branch in Github web I get a 404 on that URL

**Comment by ann0see** [comment]
Argh. Sorry. It's here: https://github.com/corrados/jamulus/blob/master/windows/translate/en.nsi

**Comment by ann0see** [comment]
@gilgongo I opened a PR with your proposal now

**Comment by ann0see** [comment]
Can we close this now?

**Comment by gilgongo** [comment]
Yes I think so.

**Comment by ann0see** [comment]
Closed.
